kafka_metric/getmessages_kafka_metric.py

```diff
@@ -651,12 +683,19 @@ def get_json_size_bytes(json_data):
     return getsizeof(json.dumps(json_data))
 
 
-def make_safe_project_string(project):
+def make_safe_project_string(project, project_mapping_dict):
     """ make a safe project name string """
     # strip underscores
-    project = PIPE.sub('', project)
-    project = PROJECT_ALNUM.sub('-', project)
-    return project
+    # project = PIPE.sub('', project)
+    # project = PROJECT_ALNUM.sub('-', project)
+    project_list = project.split('|')[1:-1]
```

Comment:
This is not a generic implementation and requires the project field to be surrounded by a pipe symbol. This needs to be modified to be generic/ configurable.